polar-core/src/data_filtering.rs

```diff
@@ -1158,3 +1223,447 @@ mod test {
         }
     }
 }
+
+// NEW DATA FILTERING
+#[derive(PartialEq, Debug, Serialize, Clone)]
+pub struct Proj(Rc<DataFilter>, String);
+#[derive(Debug)]
+pub enum PreDatum {
```

Comment:
Not too clear on what this is compared to `Datum`

Reply:
It's a `Datum` with some extra information attached to each `Field` case that we use to construct joins. Kind of an intermediate representation. It gets reduced to a plain `Datum` in the final output, hence the name.

Reply:
Got it. So, for `x.a.b.c = 1`, we'd have:

`PreDatum::Field(Proj(?, "c"), DataSource("foo", ["a", "b"]))`

I'm not clear on what the projection's source would be.